.jenkinsci/test-launcher.groovy

```diff
@@ -1,23 +1,42 @@
 #!/usr/bin/env groovy
 
+enum TestTypes {
+	module, integration, system, cmake, regression, benchmark, framework	
+}
+
+// format the enum elements output like "(val1|val2|...|valN)*"
+def printRange(start, end) {
+  def output = ""
+  for (type in start..end) {
+    output = [output, (type.name() != start.toString() ? "|" : ""), type.name()].join('')
```

Comment:
`join()` already implements the logic of not putting delimeter after the last element in the list.

Reply:
@bakhtin `join()` cannot work with `enum`. So we can either do as I proposed in the commit, either transform `enum` to `string collection/tuple` and use `join()` which is even more complex